### ISSUE TITLE
Adding a bit of order to the panel (Fasteners_ToggleThreadType)

### ISSUE THREAD

**Issue by belinski-roman** [shaise/FreeCAD_FastenersWB#203]
I see that the number of fasteners in your workbench is increasing and that's good. I want to propose an innovation that brings a little order to the panel. What if we add a switch button to change the two modes metric thread/inch thread (by the type of how it is done in the sketch workbench - buttons that perform Scketcher_ToggleConstruction and Scketcher_ToggleDrivingConstraint). So it would be possible to reduce and fix the number of buttons in the mode when all types of fasteners are displayed without a submenu. And if there is no corresponding analog fastener for the inch thread, then this button can be left gray.

**Comment by FreeCAD-Tools** [comment]
I agree need some kind of solution. May be a panel with the ability to hide and show certain types of fasteners:

![изображение](https://user-images.githubusercontent.com/90561109/194703488-13aa8dcc-7ca3-4779-bb9f-db1003de49d5.png)

**Comment by belinski-roman** [comment]
Yes, I think it's a good idea. It is necessary to think about how best to do this - in the form of a filter or as a set of checkboxes. You can also consider the option of a full-fledged sidebar instead of a workbench. This would make it possible to organize the arrangement of fastener types in a tree view of catalogs. But this is probably a topic that goes beyond the conversation about fasteners. If you implement such a thing, it is already in the context of creating a library of elements. Sincerely, Roman Belinsky.

**Comment by luzpaz** [comment]
@shaise thoughts?

**Comment by shaise** [comment]
Right now there is no easy way to do this, but changing the toolbar grouping to "Drop down buttons" can help quite a bit.
![image](https://github.com/shaise/FreeCAD_FastenersWB/assets/5813568/48d7390b-ee5d-43c6-9c4a-63a9200ec0cc)

**Comment by tobiasfalk** [comment]
What about a Dialog that pops up?
Where you then select your type, diameter and so on?
Maybe more then one Dialog, one for Fasteners, one for nut and so on.

**Comment by tobiasfalk** [comment]
This is a very crude sketch of how such a wind could be structured, in this example for a fastener's for other categories like nuts or Retaining Rings it would need adjustment
![WhatsApp Bild 2023-08-02 um 20 42 05](https://github.com/shaise/FreeCAD_FastenersWB/assets/56540480/a82e9944-a2d9-4f01-b6cd-3bcb15c812b2)

**Comment by berberic2** [comment]
related: https://forum.freecad.org/viewtopic.php?p=700019#p700019
